### PR TITLE
Add integrity test for slash defer duration

### PR DESCRIPTION
**Pull request by kianenigma**
A better way of doing: https://github.com/paritytech/polkadot/pull/1504

An optional companion _could_ remove the tests from polkadot repo. 

polkadot companion: https://github.com/paritytech/polkadot/pull/1523

### CODE REVIEW

**Comment by bkchr** [comment]
@kianenigma Tests are not happy.

**Comment by kianenigma** [comment]
This cannot be merged as it is because the despite the two values over which we are asserting are constant, they are written to storage, and thus cannot be queried outside externalities provided environment. 

https://github.com/paritytech/substrate/issues/6687 

Could potentially fix it, or some other way to allow these values to be queried both from storage and from memory.

File: frame/staking/src/lib.rs [hunk 1]
```diff
@@ -1460,9 +1460,14 @@ decl_module! {
...
-			assert!(
-				T::SlashDeferDuration::get() < T::BondingDuration::get() || T::BondingDuration::get() == 0,
-				"As per documentation, slash defer duration should be less than bonding duration.",
+			sp_io::TestExternalities::new_empty().execute_with(||
+				assert!(
+					T::SlashDeferDuration::get() < T::BondingDuration::get() || T::BondingDuration::get() == 0,
+					"As per documentation, slash defer duration ({}) should be less than bonding duration ({}).",
+					T::SlashDeferDuration::get(),
+					T::BondingDuration::get(),
+
```
Comment (kianenigma) [inline frame/staking/src/lib.rs:1469]:
```suggestion
```

**Comment by kianenigma** [comment]
bot merge

**Pull request merged**